### PR TITLE
add support for specifying latest version in cluster config

### LINKED ISSUE #2154: create cluster: unable to use "latest" as version
Looking at
https://github.com/weaveworks/eksctl/blob/master/pkg/ctl/create/utils.go#L28-L30
I thought I should be able to pass --version="latest" or
```yaml
metadata:
  name: ${USER}-eksctl
  region: ${AWS_REGION}
  version: latest
```
but when I do I get
```
Error: invalid version, supported values: 1.13, 1.14, 1.15, 1.16
```

Any idea what I'm missing?

### PR DESCRIPTION
**Pull request by aclevername**
### Description

Closes #2154 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: integration/tests/addons/addons_test.go [hunk 1]
```diff
@@ -75,6 +76,11 @@ var _ = Describe("(Integration) [EKS Addons test]", func() {
 				WithStdin(bytes.NewReader(data))
 			Expect(cmd).To(RunSuccessfully())
 
+			rawClient = getRawClient(clusterName)
+			serverVersion, err := rawClient.ServerVersion()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(serverVersion).To(ContainSubstring(api.LatestVersion))
+
```
Comment (aclevername) [inline integration/tests/addons/addons_test.go:83]: adding a *new* integration test felt like extreme overkill, so I picked a test that already created a cluster and had it use `latest` :shrug: doesn't feel super clean, but better than adding a new suite

File: pkg/ctl/create/cluster.go [hunk 2]
```diff
@@ -131,6 +131,9 @@ func doCreateCluster(cmd *cmdutils.Cmd, ngFilter *filter.NodeGroupFilter, params
 	if cfg.Metadata.Version == "" || cfg.Metadata.Version == "auto" {
 		cfg.Metadata.Version = api.DefaultVersion
 	}
+	if cfg.Metadata.Version == "latest" {
+		cfg.Metadata.Version = api.LatestVersion
+	}
```
Comment (aclevername) [inline pkg/ctl/create/cluster.go:136]: this is the only place where this seems to matter, like how we accept `version: ""` for falling back to the default version

File: integration/tests/addons/addons_test.go [hunk 3]
```diff
@@ -75,6 +76,11 @@ var _ = Describe("(Integration) [EKS Addons test]", func() {
 				WithStdin(bytes.NewReader(data))
 			Expect(cmd).To(RunSuccessfully())
 
+			rawClient = getRawClient(clusterName)
+			serverVersion, err := rawClient.ServerVersion()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(serverVersion).To(ContainSubstring(api.LatestVersion))
```
Comment (cPu1) [inline integration/tests/addons/addons_test.go:82]: Minor nit:
```suggestion
			Expect(serverVersion).To(HavePrefix(api.LatestVersion))
```

Comment (cPu1) [inline integration/tests/addons/addons_test.go:82]: Minor nit:
```suggestion
			Expect(serverVersion).To(HavePrefix(api.LatestVersion))
```

**Review by cPu1 — APPROVED** [review]

**Review by cPu1 — APPROVED** [review]

**Review by Himangini — APPROVED** [review]
👌🏻

**Review by Skarlso — APPROVED** [review]
and_my_axe

**Pull request merged**